src/components/ShoppingList/ShoppingList.js

```diff
@@ -62,6 +62,90 @@ function ShoppingList({ listId }) {
     setFilter(e.target.value);
   };
 
+  const getDaysToPurchase = (item) => {
+    let nextPurchaseDate;
+    if (item.lastPurchaseDate?.seconds) {
+      // if the item has been purchased before, next purchase date is [purchaseInterval] days from the lastPurchaseDate
+      nextPurchaseDate = DateTime.fromSeconds(
+        item.lastPurchaseDate.seconds,
+      ).plus({ days: item.purchaseInterval });
+    } else if (item.createdAt?.seconds) {
+      // if there's no purchase history, estimate it will be bought [purchaseInterval] days from when item was created
+      // (user provides this info at item creation)
+      nextPurchaseDate = DateTime.fromSeconds(item.createdAt.seconds).plus({
+        days: item.purchaseInterval,
+      });
+    } else return null;
+
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+    const daysRemaining = nextPurchaseDate.diff(currentDate, ['days']);
+    return Math.round(daysRemaining.as('days'));
+  };
+
+  const isItemInactive = (item) => {
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+
+    if (item.lastPurchaseDate?.seconds) {
+      // if an item has been purchased before, compare the time elapsed since lastPurchaseDate to the purchaseInterval
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.lastPurchaseDate.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else if (item.createdAt?.seconds) {
+      // if an item has never been purchased, do the same calculating using createdAt date instead of lastPurchaseDate
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.createdAt.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else return null;
+  };
+
+  const sortListItems = (itemA, itemB) => {
+    // if items being compared are both inactive, or neither is inactive, sort by days until next purchase
```

Comment:
I see it in your PR. It would be helpful if the explanation were included as a comment as well! The PR explanations are quite helpful. In general, it might be good to make sure they're included as comments as well - even if in short form.